website/docs/r/virtual_network_gateway_connection.html.markdown

```diff
@@ -248,8 +247,44 @@ The following arguments are supported:
 * `enable_bgp` - (Optional) If `true`, BGP (Border Gateway Protocol) is enabled
     for this connection. Defaults to `false`.
 
+* `use_policy_based_traffic_selectors` - (Optional) If `true`, policy-based traffic
+    selectors are enabled for this connection. Enabling policy-based traffic
+    selectors requires an `ipsec_policy` block. Defaults to `false`.
+
+* `ipsec_policy` (Optional) A `ipsec_policy` block which is documented below.
+    Only a single policy can be defined for a connection. For details on
+    custom policies refer to [the relevant section in the Azure documentation](https://docs.microsoft.com/en-us/azure/vpn-gateway/vpn-gateway-ipsecikepolicy-rm-powershell).
+
 * `tags` - (Optional) A mapping of tags to assign to the resource.
 
+The `ipsec_policy` block supports:
+
+* `dh_group` - (Required) The DH group used in IKE phase 1 for initial SA. Valid
+    options are `DHGroup1`, `DHGroup14`, `DHGroup2`, `DHGroup2048`, `DHGroup24`,
+    `ECP256`, `ECP384`, or `None`.
+
+* `ike_encryption` - (Required) The IKE encryption algorithm. Valid
+    options are `AES128`, `AES192`, `AES256`, `DES`, or `DES3`.
+
+* `ike_integrity` - (Required) The IKE integrity algorithm. Valid
+    options are `MD5`, `SHA1`, `SHA256`, or `SHA384`.
+
+* `ipsec_encryption` - (Required) The IPSec encryption algorithm. Valid
+    options are `AES128`, `AES192`, `AES256`, `DES`, `DES3`, `GCMAES128`, `GCMAES192`, `GCMAES256`, or `None`.
+
+* `ipsec_integrity` - (Required) The IPSec integrity algorithm. Valid
+    options are `GCMAES128`, `GCMAES192`, `GCMAES256`, `MD5`, `SHA1`, or `SHA256`.
+    
+* `pfs_group` - (Required) The DH group used in IKE phase 2 for new child SA.
+    Valid options are `ECP256`, `ECP384`, `PFS1`, `PFS2`, `PFS2048`, `PFS24`,
+    or `None`.
+
+* `sa_datasize` - (Optional) The IPSec SA payload size in KB. Must be at least
+    `1024` KB. Defaults to `102400000` KB.
```

Comment:
given the value can stretch so high, would it be worth converting this value to/from MB?

Reply:
I would recommend to keep KBytes as a unit here for two reasons: 1) In the Azure docs this values is consistently described in this unit (https://docs.microsoft.com/en-us/azure/vpn-gateway/vpn-gateway-vpn-faq#ipsecike) and 2) it is quite common among VPN vendors to configure this in Kbytes (e.g https://www.cisco.com/c/en/us/td/docs/security/security_management/cisco_security_manager/security_manager/4-1/user/guide/CSMUserGuide_wrapper/vpipsec.html#pgfId-959357)

Reply:
makes sense 👍